### PR TITLE
Add DDE support in `System`

### PR DESCRIPTION
**Pull request by YingboMa**
```julia
@parameters p0=0.2 p1=0.2 q0=0.3 q1=0.3 v0=1 v1=1 d0=5 d1=1 d2=1 beta0=1 beta1=1
@variables t x₀(t) x₁(t) x₂(..)
tau = 1
D = Differential(t)
eqs = [D(x₀) ~ (v0 / (1 + beta0 * (x₂(t - tau)^2))) * (p0 - q0) * x₀ - d0 * x₀
    D(x₁) ~ (v0 / (1 + beta0 * (x₂(t - tau)^2))) * (1 - p0 + q0) * x₀ +
            (v1 / (1 + beta1 * (x₂(t - tau)^2))) * (p1 - q1) * x₁ - d1 * x₁
    D(x₂(t)) ~ (v1 / (1 + beta1 * (x₂(t - tau)^2))) * (1 - p1 + q1) * x₁ - d2 * x₂(t)]
@named sys = System(eqs)
prob = DDEProblem(sys,
    [x₀ => 1.0, x₁ => 1.0, x₂(t) => 1.0],
    tspan,
    constant_lags = [tau])
sol_mtk = solve(prob, alg, reltol = 1e-7, abstol = 1e-10)
@test sol_mtk.u[end] ≈ sol.u[end]
prob2 = DDEProblem(sys,
    [x₀ => 1.0 - t * q1, x₁ => 1.0 - t * q1, x₂(t) => 1.0 - t * q1],
    tspan,
    constant_lags = [tau])
sol2_mtk = solve(prob2, alg, reltol = 1e-7, abstol = 1e-10)
```
now works. See `test/dde.jl` for more details.

### CODE REVIEW

File: src/systems/diffeqs/abstractodesystem.jl [hunk 1]
```diff
@@ -802,6 +874,55 @@ function DiffEqBase.DAEProblem{iip}(sys::AbstractODESystem, du0map, u0map, tspan
     end
 end
 
+function DiffEqBase.DDEProblem(sys::AbstractODESystem, args...; kwargs...)
+    DDEProblem{true}(sys, args...; kwargs...)
+end
+function DiffEqBase.DDEProblem{iip}(sys::AbstractODESystem, u0map = [],
+    h = (u, p) -> zeros(length(states(sts))),
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/abstractodesystem.jl:881]: bold assumption. yeah we'll need to have some way to give a history function.

Comment (YingboMa) [reply src/systems/diffeqs/abstractodesystem.jl:881]: Since it's a positional argument before something with a default, it must also have a default.

**Comment by ChrisRackauckas** [comment]
What is your planned representation for passing symbolic history functions? Just if the default value is a symbolic expression?

**Comment by YingboMa** [comment]
Yeah, since `h(p, 0)` is just `u0`, right?

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**